test/with_dummyserver/test_connection.py

```diff
@@ -33,6 +34,32 @@ def test_returns_urllib3_HTTPResponse(pool: HTTPConnectionPool) -> None:
     assert isinstance(response, HTTPResponse)
 
 
+hook_reported = False
+
+
+def test_audit_event(pool: HTTPConnectionPool) -> None:
+    # Audit hooks are only available in versions >= 3.8
+    if not hasattr(sys, "audit"):
+        assert True
+        return
+
+    def _hook(event: str, _: tuple[typing.Any]) -> None:
+        global hook_reported
+        if event == "http.client.connect":
+            hook_reported = True
+
+    sys.addaudithook(_hook)  # type: ignore
+
+    conn = pool._get_conn()
+
+    method = "GET"
```

Comment:
nit: No need to store `method` and `path`, these can be inlined into `conn.request(...)`.